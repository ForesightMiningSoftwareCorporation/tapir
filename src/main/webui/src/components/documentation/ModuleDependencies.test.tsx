import { render } from "@testing-library/react";
import ModuleDependencies from "./ModuleDependencies";

describe("<ModuleDependencies /> spec", () => {
  it("renders the ModuleDependencies", () => {
    const container = render(
      <ModuleDependencies
        modules={[
          { source: "terraform-aws-modules/security-group/aws", version: "" },
        ]}
        providers={[{ name: "aws" }]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
