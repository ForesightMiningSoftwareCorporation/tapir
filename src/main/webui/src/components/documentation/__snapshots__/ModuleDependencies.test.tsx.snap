// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleDependencies /> spec renders the ModuleDependencies 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Module Dependencies
      </h4>
      <p
        class="MuiTypography-root MuiTypography-body1 css-y9smp5-MuiTypography-root"
      >
         
        Dependencies are external modules that this module references. A module is considered external if it isn't within the same repository.
      </p>
      <div
        class="MuiStack-root css-149jpez-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <strong>
              terraform-aws-modules/security-group/aws
            </strong>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Version:
               
              <em>
                latest
              </em>
            </p>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-1hyyq6u-MuiDivider-root"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Provider Dependencies
      </h4>
      <p
        class="MuiTypography-root MuiTypography-body1 css-y9smp5-MuiTypography-root"
      >
        Providers are Terraform plugins that will be automatically installed, if available on the Terraform Registry.
      </p>
      <div
        class="MuiStack-root css-bxhar4-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <strong>
              aws
            </strong>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Version:
               
              <em>
                latest
              </em>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Module Dependencies
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 css-y9smp5-MuiTypography-root"
    >
       
      Dependencies are external modules that this module references. A module is considered external if it isn't within the same repository.
    </p>
    <div
      class="MuiStack-root css-149jpez-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <strong>
            terraform-aws-modules/security-group/aws
          </strong>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Version:
             
            <em>
              latest
            </em>
          </p>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-1hyyq6u-MuiDivider-root"
    />
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Provider Dependencies
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 css-y9smp5-MuiTypography-root"
    >
      Providers are Terraform plugins that will be automatically installed, if available on the Terraform Registry.
    </p>
    <div
      class="MuiStack-root css-bxhar4-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <strong>
            aws
          </strong>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Version:
             
            <em>
              latest
            </em>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
